lib/rack/test/utils.rb

```diff
@@ -7,12 +7,16 @@ module Utils # :nodoc:
       def build_nested_query(value, prefix = nil)
         case value
         when Array
-          value.map do |v|
-            unless unescape(prefix) =~ /\[\]$/
-              prefix = "#{prefix}[]"
-            end
-            build_nested_query(v, "#{prefix}")
-          end.join("&")
+          if value.empty?
+            "#{prefix}=#{escape(value)}"
```

Comment:
This didn't work in my case(rack will parse it to '[]')
Here is my working solution
`"#{prefix}[]="